### PR TITLE
Introduce commit guidelines

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Add commit guidelines documenting the commit process and providing a framework for writing commit messages that are easy to comprehend.

/cc @dcos/frontend

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,112 @@
...
+* **docs** - use this one to indicate documentation adjustments and improvements 
+* **chore** - use this type for _maintainance_ commits e.g. removing old files 
+* **style** - use this one for commits that fix formatting and linting errors
+* **refactor** - 
+  use this type for adjustments to improve maintainability or performance
+* **test** - use this one for commits that add new tests
+
+### Scope
+
+The scope can be anything specifying the place of the committed change.
```
Comment (nLight) [inline CONTRIBUTING.md]: We've discussed to use filename when it's a single file change and module/package/page name for features etc.
We could expand this a little bit based on this discussion

Comment (MatApple) [inline CONTRIBUTING.md]: I agree that it would be nice to have this as specific as possible :)

**Comment by MatApple** [comment]
LGTM after @nLight's comment so it gets my label 👍

**Comment by weblancaster** [comment]
@orlandohohmeier should we close #2003 in favor of this one? then I add what is needed on top of yours when it's merged.

**Comment by orlandohohmeier** [comment]
@weblancaster It probably makes sense to close one of the PRs and then _consolidate_ the changes.  Let's discuss next week how we land these contributing updates as I planned to document PRs and Issues as well.

File: CONTRIBUTING.md (hunk 1 above)
Comment (orlandohohmeier) [inline CONTRIBUTING.md]: 💯 That's a great idea !

**Comment by weblancaster** [comment]
I have closed #2003 in favor of this PR.

Can we make the versions of node/npm we need more explicit removing the `(and above)`, at first I thought above would work as documented and tried installing the latest LTS version.

What do you think about adding the changes below to `TESTING.md`?

### Running Unit Tests

Make sure you have Node 4.4.x and NPM 3.9.x and your packages are up to date by running `npm install`, and then running tests with:

```sh
npm test
```

To watch for changes while still working on code:

```sh
npm run test:watch
```
You can also run specific files:

```sh
npm test fileName
# or
npm run test:watch fileName
```

**Comment by orlandohohmeier** [comment]
@nLight I'v improved the scope documentation as suggested. Please check and verify my changes.

**Comment by orlandohohmeier** [comment]
@weblancaster I love you suggested but would land them in separate PRs as I wouldn't to change the scope of this one.

**Comment by nLight** [comment]
run integration tests

**Comment by nLight** [comment]
Feel free to merge once tests pass!

**Comment by mlunoe** [comment]
@weblancaster Please make sure to delete branches after merging, so we are not left with a bunch of branches in the repository.

Great work @orlandohohmeier!

**Pull request merged**